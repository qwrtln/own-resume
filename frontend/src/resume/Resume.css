.resume {
  margin: 5%;
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-gap: 1rem;
}

.basics__name-section {
  grid-column: 1 / 7;
}

.basics__contact-section {
  grid-column: 9 / 13;
}

.basics__side-section {
  grid-row: 2 / 5;
  grid-column: 1 / 5;
}

.basics__contact-field-label {
  font-weight: bold;
  font-size: 1.2rem;
  margin: 0.5rem;
}

.basics > div {
  padding: 1.5rem;
  border-radius: 1rem;
  text-align: left;
  background: #aaa;
}

.basics-photo {
  width: 100%;
}

.basics__name-section > h1 {
  font-size: 2.4rem;
  border-bottom: 1px solid #666;
  padding: 0.2rem;
  margin: 0;
}

.basics__name-section > h2 {
  color: #666;
  text-transform: uppercase;
  font-size: 1.6rem;
  padding: 0.4rem;
  margin: 0;
}

.workplaces {
  grid-row: 2;
  grid-column: 6 / 13;
  grid-template-columns: 1fr 4fr;
}

.workplaces > h2 {
  font-size: 1.6rem;
  border-bottom: 1px solid #666;
}

.workplace > h3 {
  font-size: 1.6rem;
}
