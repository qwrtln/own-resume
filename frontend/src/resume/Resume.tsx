import React from 'react';
import "./Resume.css"
import {ResumeModel} from "./ResumeModel";
import Basics from "./Basics";
import Workplaces from "./Workplaces";

const Resume: React.FC<ResumeModel> = (props) => (
    <div>
        <Basics {...props.basics}/>
        <Workplaces workplaces={props.work}/>
    </div>
);

export default Resume;
