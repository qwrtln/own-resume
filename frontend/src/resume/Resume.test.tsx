import React from "react";
import { render } from "@testing-library/react";
import Resume from "./Resume";

test("render resume", () => {
   let resume = render(
    <Resume
      basics={{ name: "Johnny Python", email: "" }}
      work={[
        {
          company: "Company 1",
        },
      ]}
    />
  );

  expect(resume).toMatchSnapshot();
});
