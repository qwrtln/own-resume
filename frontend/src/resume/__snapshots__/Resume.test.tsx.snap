// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render resume 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="resume"
      >
        <div
          class="basics__name-section"
        >
          <h1>
            Johnny Python
          </h1>
          <h2 />
        </div>
        <div
          class="basics__contact-section"
        >
          <div>
            <span
              class="basics__contact-field-label"
            >
              Mail:
            </span>
            <span />
          </div>
          <div>
            <span
              class="basics__contact-field-label"
            >
              Phone:
            </span>
            <span />
          </div>
          <div>
            <span
              class="basics__contact-field-label"
            >
              Website:
            </span>
            <span />
          </div>
        </div>
        <div
          class="basics__contact-section"
        />
        <div
          class="basics__side-section"
        >
          <img
            class="basics-photo"
            src="photo.jpg"
          />
          <h3>
            About me
          </h3>
        </div>
        <div
          class="workplaces"
        >
          <h2>
            Workplaces:
          </h2>
          <div
            class="workplace"
          >
            <h3>
              Company 1
            </h3>
            <h4 />
            <div>
              <div>
                <h5>
                  Summary: 
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="resume"
    >
      <div
        class="basics__name-section"
      >
        <h1>
          Johnny Python
        </h1>
        <h2 />
      </div>
      <div
        class="basics__contact-section"
      >
        <div>
          <span
            class="basics__contact-field-label"
          >
            Mail:
          </span>
          <span />
        </div>
        <div>
          <span
            class="basics__contact-field-label"
          >
            Phone:
          </span>
          <span />
        </div>
        <div>
          <span
            class="basics__contact-field-label"
          >
            Website:
          </span>
          <span />
        </div>
      </div>
      <div
        class="basics__contact-section"
      />
      <div
        class="basics__side-section"
      >
        <img
          class="basics-photo"
          src="photo.jpg"
        />
        <h3>
          About me
        </h3>
      </div>
      <div
        class="workplaces"
      >
        <h2>
          Workplaces:
        </h2>
        <div
          class="workplace"
        >
          <h3>
            Company 1
          </h3>
          <h4 />
          <div>
            <div>
              <h5>
                Summary: 
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
